lnrpc/routerrpc/router_server.go

```diff
@@ -1358,7 +1358,10 @@ func (s *Server) trackPayment(subscription routing.ControlTowerSubscriber,
 
 	// Otherwise, we will log and return the error as the stream has
 	// received an error from the payment lifecycle.
-	log.Errorf("TrackPayment got error for payment %v: %v", identifier, err)
+	if err != nil {
```

Comment:
can be removed now because the err check is done beforehand.

Reply:
In this function? In the switch statement, the `err != nil` case will hit `default:`, which will fall through to this case.

Reply:
yeah the latest version from another PR added `case err == nil` to catch the nil err case, so it's no longer needed.